### PR TITLE
Catch exceptions and show error on command line

### PR DESCRIPTION
**Pull request by bennuttall**
@alexsavio does this look sensible to you?

### CODE REVIEW

File: scripts/pyjoke [hunk 1]
```diff
@@ -39,7 +39,14 @@ def main():
         parser.error(str(exc.message))
         exit(-1)
 
-    joke = get_joke(language=args.language, category=args.category)
+    try:
+        joke = pyjokes.get_joke(language=args.language, category=args.category)
+    except pyjokes.LanguageNotFoundError:
+        print('No such language %s' % args.language)
```
Comment (alexsavio) [inline scripts/pyjoke]: I guess this error message would make more sense: `'No jokes found in language {}'.format(args.language)`.
I mean, if the language is a existing one you can't say there is no such a language, right?
You could also invite people to add jokes in the language.

P.S.: I don't care if you use the `format` function or the `%`.

File: scripts/pyjoke [hunk 2]
```diff
@@ -39,7 +39,14 @@ def main():
...
+        exit(-1)
+    except pyjokes.CategoryNotFoundError:
+        print('No such category %s' % args.category)
```
Comment (alexsavio) [inline scripts/pyjoke]: `'There are no jokes for category {}'.format(args.category)`. ??
Well, you are the one immersed in the RP English. You should know. :smiley:

**Comment by bennuttall** [comment]
Thanks. Yeah makes sense.

I'm a `%` guy.

Maybe we need to expand to `There are no jokes in category X for language Y` or something?

**Comment by alexsavio** [comment]
To expand as you say we would need the set of valid options for language and category and check that before querying. Mainly because, for example, in German we have this:
```
jokes_de = {
    'neutral': neutral,
    'all': neutral,
}
```
So a call to `get_joke('de', 'chuck')` would raise `CategoryNotFoundError` but not because `chuck` does not exist as a category but because `chuck` is not present in `pyjokes_de.jokes_de`. In order to know that we need beforehand the set of valid options (not only in the docstrings).

P.S.: Isn't `pyjokes_de.jokes_de` a little bit redundant?

**Pull request merged**